### PR TITLE
Update changelog for v1.0.0-alpha.17

### PR DESCRIPTION
**Pull request by conradoplg**
## Motivation

Update changelog for v1.0.0-alpha.17 release.

## Review

Blocked on https://github.com/ZcashFoundation/zebra/pull/2750

Needs to be updated with upcoming PRs to be merged

### Reviewer Checklist

  - [x] Release date is correct
  - [x] Contains recently merged PRs

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,24 @@ All notable changes to Zebra will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org).
 
+## [Zebra 1.0.0-alpha.17](https://github.com/ZcashFoundation/zebra/releases/tag/v1.0.0-alpha.17) - 2021-09-10
```
Comment (mpguerra) [inline CHANGELOG.md]:
```suggestion
## [Zebra 1.0.0-alpha.17](https://github.com/ZcashFoundation/zebra/releases/tag/v1.0.0-alpha.17) - 2021-09-14
```

Comment (mpguerra) [reply CHANGELOG.md]: Assuming we will tag the release today :)

**Review by mpguerra — CHANGES_REQUESTED** [review]
#2747 and #2741 have been merged since this PR was started, they will need to be added in before merge

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,24 @@ All notable changes to Zebra will be documented in this file.
...
+### Added
+
+- Monitor changes to the chain tip, decide if sync reached tip (#2695, #2685,
+  #2686, #2715, #2721, #2722)
+- Only enable the mempool crawler after synchronization reaches the chain tip (#2667)
+- Reply to requests for transactions IDs in the mempool (#2720)
+- Reply to requests for transactions in the mempool, given their IDs (#2725)
+- Download and verify gossiped transactions (#2679, #2727, #2718)
+
+### Security
```
Comment (mpguerra) [inline CHANGELOG.md:30]:
```suggestion
#### Documentation

- Document consensus rules for version group IDs (#2719)
- Specify Zebra Client will only support Unified Addresses (#2706) 

### Fixed

-  Stop calculating transaction hashes twice in the checkpoint verifier (#2696) 

### Security
```

Comment (mpguerra) [reply CHANGELOG.md:30]: These might be interesting enough to include in the changelog also

**Review by dconnolly — APPROVED** [review]

**Pull request merged**